src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -292,6 +292,11 @@ public void ParseStream(Stream stream, bool metadataOnly = false)
                         case JpegConstants.Markers.SOS:
                             if (!metadataOnly)
                             {
+                                if (this.Frame is null)
+                                {
+                                    throw new ImageFormatException("no readable SOF marker found.");
```

Comment:
Can you change this to say "No readable SOFn (Start Of Frame) marker found" and move the check to `ProcessStartOfScanMarker()`

Reply:
sure, done.